severe-weather/loader.py

```diff
@@ -36,86 +37,65 @@ def __call__(self, batch):
                 stds = torch.tensor(self.stds[m], device=image.device)
             else:
                 msg = (
-                    f"Expected batch with 4 dimensions (B, C, H, W), sample with 2 dimensions (H, W) "
-                    f"or a single channel, but got {len(image.shape)}"
+                    f'Expected batch with 4 dimensions (B, C, H, W), sample with 2 dimensions (H, W) '
+                    f'or a single channel, but got {len(image.shape)}'
                 )
                 raise Exception(msg)
 
-            batch["image"][m] = (image - means) / stds
+            batch['image'][m] = (image - means) / stds
         return batch
 
 
 # https://torchgeo.readthedocs.io/en/latest/tutorials/contribute_non_geo_dataset.html
 class SatChipDataset(NonGeoDataset):
-    def __init__(self, label_path, s2_path, rtc_path, transforms=None, split='train'):
+    def __init__(self, chip_path, transforms=None, split='train'):
         if not transforms:
             self.transforms = ToTensorV2()
         else:
             self.transforms = transforms
 
         self.slice_range = slice(3, 259)
-
-        label_ds = _load_ds(label_path)
-        s2_ds = _load_ds(s2_path)
-        rtc_ds = _load_ds(rtc_path)
-
-        # TODO: enforce this in satchip as well
-        band_order = [
-            'COASTAL',
-            'BLUE',
-            'GREEN',
-            'RED',
-            'REDEDGE1',
-            'REDEDGE2',
-            'REDEDGE3',
-            'NIR',
-            'NIR08',
-            'NIR09',
-            'SWIR16',
-            'SWIR22',
-        ]
-
-        self.label_ds = label_ds
-        samples = self.label_ds.sample
-
-        self.s2_ds = s2_ds.sel(sample=samples).reindex(sample=samples, band=band_order)
-        self.rtc_ds = rtc_ds.sel(sample=samples).reindex(sample=samples, band=['VV', 'VH'])
+        self.label_path = chip_path / split / 'LABEL'
+        self.rtc_path = chip_path / split / 'S1RTC'
+        self.s2_path = chip_path / split / 'S2L2A'
+        self.chip_names = ['_'.join(x.name.split('_')[-4:]).split('.')[0] for x in self.label_path.glob('*.zarr.zip')]
+        self.chip_names = sorted(self.chip_names)
+        labels = list(self.label_path.glob('*.zarr.zip'))
+        rtcs = list(self.rtc_path.glob('*.zarr.zip'))
+        s2s = list(self.s2_path.glob('*.zarr.zip'))
+        self.chip_list = []
+        for name in self.chip_names:
+            label = [x for x in labels if name in x.name][0]
+            rtc = [x for x in rtcs if name in x.name][0]
+            s2 = [x for x in s2s if name in x.name][0]
+            self.chip_list.append({'LABEL': label, 'S1RTC': rtc, 'S2L2A': s2})
```

Comment:
How do you feel about remove member variables that are not used outside of the constructor?
```suggestion
        label_path = chip_path / split / 'LABEL'
        rtc_path = chip_path / split / 'S1RTC'
        s2_path = chip_path / split / 'S2L2A'
        chip_names = ['_'.join(x.name.split('_')[-4:]).split('.')[0] for x in label_path.glob('*.zarr.zip')]
        chip_names = sorted(chip_names)
        labels = list(label_path.glob('*.zarr.zip'))
        rtcs = list(rtc_path.glob('*.zarr.zip'))
        s2s = list(s2_path.glob('*.zarr.zip'))
        self.chip_list = []
        for name in chip_names:
            label = [x for x in labels if name in x.name][0]
            rtc = [x for x in rtcs if name in x.name][0]
            s2 = [x for x in s2s if name in x.name][0]
            self.chip_list.append({'LABEL': label, 'S1RTC': rtc, 'S2L2A': s2})
```

Reply:
Good idea!